rxnetty-common/src/main/java/io/reactivex/netty/threads/RxJavaEventloopScheduler.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2015 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.reactivex.netty.threads;
+
+import io.netty.channel.EventLoop;
+import io.netty.channel.EventLoopGroup;
+import io.netty.util.concurrent.Future;
+import rx.Scheduler;
+import rx.Subscription;
+import rx.annotations.Beta;
+import rx.functions.Action0;
+import rx.internal.util.SubscriptionList;
+import rx.subscriptions.Subscriptions;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A scheduler that uses a provided {@link EventLoopGroup} instance to schedule tasks. This should typically be used as
+ * a computation scheduler or any other scheduler that do not schedule blocking tasks. <p>
+ */
+@Beta
+public class RxJavaEventloopScheduler extends Scheduler {
+
+    private final EventLoopGroup eventLoopGroup;
+
+    public RxJavaEventloopScheduler(EventLoopGroup eventLoopGroup) {
+        this.eventLoopGroup = eventLoopGroup;
+    }
+
+    @Override
+    public Worker createWorker() {
+        final EventLoop eventLoop = eventLoopGroup.next();
+
+        return new Worker() {
+
+            private final SubscriptionList subs = new SubscriptionList();
```

Comment:
The use of `SubscriptionList` is problematic with delayed scheduling due to O(n) removal. A more considerate container would be a regular ˙CompositeSubscription`. If this scheduler is dominated by non-delayed tasks, you can have both `SubscriptionList` and `CompositeSubscription`. Non-delayed tasks are then added to the list and delayed are added to the composite.